kmod/src/item.c

```diff
@@ -2530,27 +2530,41 @@ void scoutfs_item_invalidate(struct super_block *sb, struct scoutfs_key *start,
 	put_pg(sb, right);
 }
 
+static unsigned long item_cache_count_objects(struct shrinker *shrink,
+					      struct shrink_control *sc)
+{
+	struct item_cache_info *cinf = container_of(shrink,
+						    struct item_cache_info,
+						    shrinker);
+	struct super_block *sb = cinf->sb;
+
+	scoutfs_inc_counter(sb, item_cache_count_objects);
+
+	/* magic numbers as we approach ~0UL :/ */
+	return min_t(u64, (u64)(cinf->lru_pages), ULONG_MAX / 2);
```

Comment:
It's annoying that there isn't help for not having the unsigned long count wander up  into the magic SHRINK_{STOP|EMPTY} values.  But here we are.  I wonder if instead of doing this raw min(ulong_max/2) in each scan, if we shouldn't have a static inline helper in util.h that returns the min of the count and LONG_MAX.  Seems cleaner, and a comment around that helper could explain what's going on.